### DISCUSSION TITLE
How to load raw PCM?

### DISCUSSION THREAD

**Discussion by jyviko** [katspaugh/wavesurfer.js#2774] [General]
Hi,
I am trying to load a raw PCM, 16Khz, 16bit, 2 ch, signed integer audio file in wavesurfer but I am not getting a waveform back nor can I play the file. I am able to render other file types correctly. Is there a way to load raw audio files? Should I change the backend? But how can I specify the sampling rate and nitrate?
Any help suggestions are appreciated.

**Comment by katspaugh** [comment]
Yes, try setting `backend: "MediaElement"`. Although if it failed to render it with the default backend, it will do so with a media element too. Can you see any decoding errors in the console? Try `wavesurfer.on('error', console.log)`.

**Reply by jyviko** [reply]
Hi @katspaugh, tried `backend: "MediaElement"` and I am getting: `Error loading media element`
With the default backend I am getting `Error decoding audiobuffer`. It makes senses as the backend doesn't know how to read the buffer. There has to be a way to input sampling rate, bit rate and endianness and type.

**Reply by katspaugh** [reply]
Sample rate can be set if you pass your own `audioContext` option. Although my understanding is, Web Audio resamples any sample rate into its own. But worth a try.

**Reply by katspaugh** [reply]
But with PCM, I'm not even sure it can decode it. It's already decoded. You basically need to insert it into a Web Audio buffer directly.

**Reply by katspaugh** [reply]
You could try something like
```
wavesurfer.load(
  'data:audio/wav;base64,UklGRigAAABXQVZFZm10IBIAAAABAAEARKwAAIhYAQACABAAAABkYXRhAgAAAAEA', // a placeholder audio file
 [YOUR_PCM_DATA_AS_FLOAT32_ARRAY],
 AUDIO_DURATION_IN_SECONDS
)
```

This theoretically should put your PCM directly into a Web Audio buffer (don't use the MediaElement backend then).

**Reply by jyviko** [reply]
I will give it a try. Thank you!
